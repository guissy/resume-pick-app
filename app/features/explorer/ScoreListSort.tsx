import React from 'react';
import styles from './ScoreList.css';

type Props = {
  name: string | React.ReactNode;
  sort: string;
  onClick: () => void;
};

const ScoreListSort: React.FC<Props> = ({ name, sort, onClick }) => {
  const [up, down] = React.useMemo(
    () =>
      name === '分数' ? ['scoreUp', 'scoreDown'] : ['workAgeUp', 'workAgeDown'],
    [name]
  );
  return (
    <button className={styles.sortBtn} type="button" onClick={onClick}>
      <span>{name}</span>
      <div className={styles.caretInit}>
        <i className={`fa fa-caret-up ${sort === up ? styles.active : ''}`} />
        <i
          className={`fa fa-caret-down ${sort === down ? styles.active : ''}`}
        />
      </div>
    </button>
  );
};

export default ScoreListSort;
