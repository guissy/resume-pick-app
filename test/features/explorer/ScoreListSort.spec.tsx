import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import ScoreListSort from '../../../app/features/explorer/ScoreListSort';

Enzyme.configure({ adapter: new Adapter() });

const defaultTestProps = {
  name: '分数' as const,
  sort: '' as const,
  onClick: () => {},
};

function setup(props = defaultTestProps) {
  // eslint-disable-next-line react/jsx-props-no-spreading
  const getWrapper = () => mount(<ScoreListSort {...props} />);
  const component = getWrapper();
  return {
    component,
    icon: component.find('i.fa'),
  };
}

describe('ScoreListSort component', () => {
  it('should match exact snapshot', () => {
    const tree = renderer
      // eslint-disable-next-line react/jsx-props-no-spreading
      .create(<ScoreListSort {...defaultTestProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match exact snapshot up', () => {
    const tree = renderer
      // eslint-disable-next-line react/jsx-props-no-spreading
      .create(<ScoreListSort {...defaultTestProps} sort="scoreUp" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match exact snapshot down', () => {
    const tree = renderer
      // eslint-disable-next-line react/jsx-props-no-spreading
      .create(<ScoreListSort {...defaultTestProps} sort="scoreDown" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call onClick', () => {
    const onClick = jest.fn();
    const { component } = setup({ ...defaultTestProps, onClick });
    component.simulate('click');
    const mouseEvent = expect.objectContaining({ type: 'click' });
    expect(onClick).toBeCalledWith(mouseEvent);
  });
});
