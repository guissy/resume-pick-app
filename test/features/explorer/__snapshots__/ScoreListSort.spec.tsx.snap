// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreListSort component should match exact snapshot 1`] = `
<button
  className="sortBtn"
  onClick={[Function]}
  type="button"
>
  <span>
    分数
  </span>
  <div
    className="caretInit"
  >
    <i
      className="fa fa-caret-up "
    />
    <i
      className="fa fa-caret-down "
    />
  </div>
</button>
`;

exports[`ScoreListSort component should match exact snapshot down 1`] = `
<button
  className="sortBtn"
  onClick={[Function]}
  type="button"
>
  <span>
    分数
  </span>
  <div
    className="caretInit"
  >
    <i
      className="fa fa-caret-up "
    />
    <i
      className="fa fa-caret-down active"
    />
  </div>
</button>
`;

exports[`ScoreListSort component should match exact snapshot up 1`] = `
<button
  className="sortBtn"
  onClick={[Function]}
  type="button"
>
  <span>
    分数
  </span>
  <div
    className="caretInit"
  >
    <i
      className="fa fa-caret-up active"
    />
    <i
      className="fa fa-caret-down "
    />
  </div>
</button>
`;
